### PR TITLE
 switch networking to external types

### PR DESCRIPTION
**Pull request by deads2k**
While trying to create a separate set of controller informers, I stumbled onto this related, low hanging external fruit.

@openshift/sig-networking 
/assign @danwinship @knobunc 

@liggitt I'll make the openshift/api pull.  How do you feel about it?  I think these are API. The construction of the annotation is a little weird, but I can accept its convenient and doesn't bind methods to types.

### CODE REVIEW

File: pkg/network/apis/network/validation/validation.go [hunk 1]
```diff
@@ -37,13 +36,6 @@ func validateIPv4(ip string) (net.IP, error) {
 	return bytes, nil
 }
 
-var defaultClusterNetwork *networkapi.ClusterNetwork
-
-// SetDefaultClusterNetwork sets the expected value of the default ClusterNetwork record
-func SetDefaultClusterNetwork(cn networkapi.ClusterNetwork) {
-	defaultClusterNetwork = &cn
```
Comment (deads2k) [inline pkg/network/apis/network/validation/validation.go]: So this was really weird.  This is a method that sets a global that was only called from a package that ran as part of the controller manager.

Thinking back, perhaps that import is accidentally called from the server.....  I'll double check

Comment (deads2k) [reply pkg/network/apis/network/validation/validation.go]: yeah, only called when you start the SDN controller which happens in a different process than the one that does the validation.

File: pkg/cmd/openshift-controller-manager/controller/interfaces.go [hunk 2]
```diff
@@ -246,8 +246,8 @@ func (b OpenshiftControllerClientBuilder) OpenshiftInternalNetworkClient(name st
 // OpenshiftInternalNetworkClientOrDie provides a REST client for the network API.
```
Comment (mfojtik) [inline pkg/cmd/openshift-controller-manager/controller/interfaces.go]: change the comment

File: pkg/cmd/openshift-controller-manager/controller/interfaces.go [hunk 3]
```diff
@@ -235,7 +235,7 @@ func (b OpenshiftControllerClientBuilder) OpenshiftInternalQuotaClientOrDie(name
 // OpenshiftInternalNetworkClient provides a REST client for the network API.
```
Comment (mfojtik) [inline pkg/cmd/openshift-controller-manager/controller/interfaces.go]: comment

File: pkg/cmd/server/kubernetes/network/network_config.go [hunk 4]
```diff
@@ -36,7 +36,7 @@ type NetworkConfig struct {
 	// Internal kubernetes shared informer factory.
 	InternalKubeInformers kinternalinformers.SharedInformerFactory
 	// Internal network shared informer factory.
```
Comment (mfojtik) [inline pkg/cmd/server/kubernetes/network/network_config.go]: not internal anymore

File: pkg/network/apis/network/validation/validation.go [hunk 5]
```diff
@@ -123,31 +115,6 @@ func ValidateClusterNetwork(clusterNet *networkapi.ClusterNetwork) field.ErrorLi
 		}
 	}
 
-	if clusterNet.Name == networkapi.ClusterNetworkDefault && defaultClusterNetwork != nil {
```
Comment (mfojtik) [inline pkg/network/apis/network/validation/validation.go]: what happened with this?

File: vendor/github.com/openshift/api/network/v1/annotations.go [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+	value := string(action)
+	if len(params) != 0 {
+		value = fmt.Sprintf("%s:%s", value, params)
+	}
+	netns.Annotations[ChangePodNetworkAnnotation] = value
+}
+
+// DeleteChangePodNetworkAnnotation removes network change intent from NetNamespace
+func DeleteChangePodNetworkAnnotation(netns *NetNamespace) {
```
Comment (mfojtik) [inline vendor/github.com/openshift/api/network/v1/annotations.go]: reminder to do the same with apps...

File: pkg/network/apis/network/validation/validation.go (hunk 5 above)
Comment (deads2k) [reply pkg/network/apis/network/validation/validation.go]: > what happened with this?

It was in my comment.  Turns out in the apiserver it was never set, so this value never matterd

**Comment by deads2k** [comment]
/hold 

for real upstreams

**Pull request merged**